pallets/phala/src/compute/basepool.rs

```diff
@@ -144,51 +147,71 @@ pub mod pallet {
 		InvalidWithdrawalAmount,
 		/// RMRK errors
 		RmrkError,
-
+		/// The Specific pid does not match to any pools
 		PoolDoesNotExist,
-
+		/// Occurs when tring to get a stakepool from a PoolProxy contains a vault or vault from a stakepool instead.
 		PoolTypeNotMatch,
-
+		/// NftId does not match any nft
 		NoAvailableNftId,
-
+		/// Occurs when pool's shares is zero
 		InvalidSharePrice,
-
+		/// Try to get a NftGuard when the nft's spinlock is active
 		AttrLocked,
 	}
 
 	#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
 	pub struct BasePool<AccountId, Balance> {
+		/// Pool ID
 		pub pid: u64,
-
+		/// The owner of the pool
 		pub owner: AccountId,
-
+		/// Total shares
+		///
+		/// It tracks the total number of shared of all the contributors. Guaranteed to be
+		/// non-dust.
 		pub total_shares: Balance,
-
+		/// It tracks the current asset value owned by the pool
+		///
+		/// Including:
+		/// 1.Freestakes of the pool (both case)
+		/// 2.Stakes bounded to the miner (stakepool case)
+		/// 3.Shares' current values owned by pool_account_id (vault case)
+		/// 4.Shares' current values which is in withdraw_queue (both case)
 		pub total_value: Balance,
-
+		/// The queue of withdraw requests
 		pub withdraw_queue: VecDeque<WithdrawInfo<AccountId>>,
-
+		/// The list of stakepools delegated by the vault
 		pub value_subscribers: VecDeque<u64>,
-
+		/// The collection_id of the pool
 		pub cid: CollectionId,
-
+		/// The account used when the pool is acting as a delegater role
+		///
+		/// Usage:
+		/// 1.Maintains freestakes of the pool with its asset account
+		/// 2.Contributes or withdraws from a stakepool (vault case)
 		pub pool_account_id: AccountId,
 	}
 
+	/// The lockguard of nft-attr
```

Comment:
```suggestion
	/// The scope guard that holds the lock of a nft for internal data security
	///
	/// When guard runs out of the scope, it will release the lock in the storage automatically.
```

Reply:
done